src/test/java/uk/gov/hmcts/ccd/domain/service/common/CompoundAccessControlServiceTest.java

```diff
@@ -1113,6 +1118,181 @@ void shouldDenyAccessWhenChildUpdatedAndNoFineGrainedACLExists() throws IOExcept
+                                            .withDelete(true)
+                                            .withUpdate(true)
+                                            .withRead(true)
+                                            .withRole(ROLE_IN_USER_ROLES).build())
+                                        .build()
+
+                                )
+                                .build()
+                        )
+                        .withComplexACL(aComplexACL()
+                            .withCreate(true)
+                            .withDelete(true)
+                            .withUpdate(true)
+                            .withRead(true)
+                            .withRole(ROLE_IN_USER_ROLES).build())
+                        .build()
+                ))
+                .build();
+
+
+            JsonNode newDataNode = getJsonNode("{\n"
+                + "  \"FieldId\": {\n"
+                + "     \"NestedField2\": [\n"
+                + "        {\n"
+                + "            \"id\": \"123\",\n"
+                + "            \"value\": {\n"
+                + "                \"SubNestedField1\": \"CollectionValue1\",\n"
+                + "                \"SubNestedField2\": \"SomethingToChanged\"\n"
+                + "            }\n"
+                + "        },\n"
+                + "        {\n"
+                + "            \"id\": \"456\",\n"
+                + "            \"value\": {\n"
+                + "                \"SubNestedField1\": \"CollectionValue2\",\n"
+                + "                \"SubNestedField2\": \"SomethingToBeIgnored\"\n"
+                + "            }\n"
+                + "        }\n"
+                + "    ]\n"
+                + "  }\n"
+                + "}");
+            JsonNode existingDataNode = getJsonNode("{\n"
+                + "  \"FieldId\": {\n"
+                + "     \"NestedField2\": [\n"
+                + "        {\n"
+                + "            \"id\": \"123\",\n"
+                + "            \"value\": {\n"
+                + "                \"SubNestedField1\": \"CollectionValue1\",\n"
+                + "                \"SubNestedField2\": \"SomethingToBeIgnored\"\n"
+                + "            }\n"
+                + "        },\n"
+                + "        {\n"
+                + "            \"id\": \"456\",\n"
+                + "            \"value\": {\n"
+                + "                \"SubNestedField1\": \"CollectionValue2\",\n"
+                + "                \"SubNestedField2\": \"SomethingToBeIgnored\"\n"
+                + "            }\n"
+                + "        }\n"
+                + "    ]\n"
+                + "  }\n"
+                + "}");
+
+            assertThat(compoundAccessControlService.hasAccessForAction(
+                newDataNode, existingDataNode, caseTypeDefinition.getCaseFieldDefinitions().get(0),
+                USER_ROLES), is(false));
```

Comment:
Think something is wrong with the set up of this test, as ACL is actually all true but access is still being denied?